atlasdb-client/src/test/java/com/palantir/atlasdb/schema/cleanup/CleanupMetadataTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.schema.cleanup;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.stream.IntStream;
+
+import org.junit.Test;
+
+import com.palantir.atlasdb.table.description.ValueType;
+
+public class CleanupMetadataTest {
+    @Test
+    public void visitorThrowsIfMethodUnimplemented() {
+        assertThatThrownBy(() -> new HashComponentsVisitor().visit(new ArbitraryCleanupMetadata()))
+                .isInstanceOf(UnsupportedOperationException.class);
+        assertThatThrownBy(() -> new HashComponentsVisitor().visit(new NullCleanupMetadata()))
+                .isInstanceOf(UnsupportedOperationException.class);
+    }
+
+    @Test
+    public void canInvokeMethodsOnVisitor() {
```

Comment:
This test may be non-trivial to debug if it fails. Does `isEqualTo` return a useful error message if it fails (expected + actual would be enough).

Reply:
Yep, can confirm it does. You'll get something of this form if things aren't equal:

```
org.junit.ComparisonFailure: 
Expected :7
Actual   :0
```

Reply:
That error is good enough for me!